### PR TITLE
[Bugfix] Fix Course Archival Issues

### PR DESCRIPTION
**Pull request by ivanchimeno**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/server.html).
- [x] Server: I documented the Java code using JavaDoc style.

### Motivation and Context

Issues were encountered while archiving courses with many exercises. We saw that multiple zip files from the exported exercises and submissions were deleted before the archive process has finished. This led to an incomplete archive file and many sentry errors since the files are no longer present. 

After some investigations, we think the root cause of the problem is the deletion scheduling which cleans up temporary files included the exported zip after x minutes have passed. 
 
### Description

We made multiple changes:

1. The export functions do not schedule the deletions anymore. This is done by the services that explicitly use the export functionality.
2. We added more exception logging and improved error messages so that we can investigate further if the course archival process still causes problems
3. We attached a small report within the archive file which tells the user how many files were not included

### Steps for Testing

The best way to test this is to have a course that has a couple of exercises of each type. Each exercise should have a couple of submissions. You could technically upload some "large files" e.g a couple of pictures that have ca. 5mb just to stress test the process a little bit.

1. Log in to Artemis
2. Navigate to Course Administration
3. Edit the course and set the date to the past
4. You should see the archive button
5. Click on archive and wait until it finishes.
6. Press the "Download Archive" button that should have appeared and download the file
7. Make sure the submissions are present as well as the template, test, and solution repositories.
8. You should also see a `report.csv` file in the archive which gives a small report
9. You might get some messages at the end of the archive about files that don't exist. This is normal because the archive process does not terminate if files are missing. It simply prints out a warning to the user and dumps those in the `exportErrors.txt` file which is also included within the archive.

### CODE REVIEW

**Review by ArtemTsurikov — APPROVED** [review]
Tested on ts1, everything worked as expected!

**Review by KiraNickel — DISMISSED** [review]
Tested on ts1, works as expected 👍🏼

**Review by willergeorg — APPROVED** [review]
Tested on TS2. Worked fine.

**Review by canyelten — DISMISSED** [review]
Tested on TS1, works as expected.

**Review by SimonEntholzer — DISMISSED** [review]
Tested locally with production dump. It takes long, but works. :+1:

**Review by FlorianGlombik — DISMISSED** [review]
Catching `Exception` is not good, but it was discussed and will be removed in a future PR. 
For now, and for ensuring that helpful error messages are printed in case the production environment does not work, since exporting courses like EIST is hard to simulate locally. - the simulation that @SimonEntholzer did during the testing session with a database dump worked as expected

**Review by NicolasUllmann — APPROVED** [review]
Tested on TS2. Everything worked as expected.

File: src/main/java/de/tum/in/www1/artemis/service/CourseExamExportService.java [hunk 1]
```diff
@@ -123,21 +139,31 @@ public CourseExamExportService(ProgrammingExerciseExportService programmingExerc
         var timestamp = ZonedDateTime.now().format(DateTimeFormatter.ofPattern("yyyyMMdd-Hmss"));
         var examDirName = exam.getId() + "-" + exam.getTitle() + "-" + timestamp;
         var cleanExamDirName = fileService.removeIllegalCharacters(examDirName);
+        List<ArchivalReportEntry> reportData = new LinkedList<>();
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/service/CourseExamExportService.java:142]: we typically use ArrayList

File: src/main/java/de/tum/in/www1/artemis/service/CourseExamExportService.java [hunk 2]
```diff
@@ -77,19 +79,33 @@ public CourseExamExportService(ProgrammingExerciseExportService programmingExerc
         var timestamp = ZonedDateTime.now().format(DateTimeFormatter.ofPattern("yyyyMMdd-Hmss"));
         var courseDirName = course.getShortName() + "-" + course.getTitle() + "-" + timestamp;
         String cleanCourseDirName = fileService.removeIllegalCharacters(courseDirName);
+        List<ArchivalReportEntry> reportData = new LinkedList<>();
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/service/CourseExamExportService.java:82]: ArrayList

**Review by KiraNickel — APPROVED** [review]
Re-approve, tested again on ts1, still works as expected

**Pull request merged**